File: dotnet/DesignPatterns/BehavioralDesignPatterns/IteratorDesignPattern/IteratorDesignPattern.Example2/Order.cs
namespace IteratorDesignPattern.Example2;

public class Order
{
    public int Id { get; set; }
    public string CustomerName { get; set; }
    public DateTime OrderDate { get; set; }
    public decimal TotalAmount { get; set; }
    public OrderStatus Status { get; set; }

    public Order(int id, string customerName, DateTime orderDate, decimal totalAmount, OrderStatus status)
    {
        Id = id;
        CustomerName = customerName;
        OrderDate = orderDate;
        TotalAmount = totalAmount;
        Status = status;
    }

    public override string ToString()
    {
        return $"Order #{Id} - {CustomerName}, {OrderDate:yyyy-MM-dd}, {TotalAmount:C}, Status: {Status}";
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/IteratorDesignPattern/IteratorDesignPattern.Example2/src/Order.ts
export enum OrderStatus {
  Pending = "Pending",
  Processing = "Processing",
  Shipped = "Shipped",
  Delivered = "Delivered",
  Cancelled = "Cancelled",
}

export class Order {
  constructor(
    public readonly id: number,
    public readonly customerName: string,
    public readonly orderDate: Date,
    public readonly totalAmount: number,
    public readonly status: OrderStatus
  ) {}

  toString(): string {
    const dateStr = this.orderDate.toISOString().slice(0, 10);
    return `Order #${this.id} - ${this.customerName}, ${dateStr}, $${this.totalAmount.toFixed(2)}, Status: ${this.status}`;
  }
}
